Fortran/gfortran/utils/update-test-config.py

```diff
@@ -0,0 +1,633 @@
+re_shouldfail = re.compile(f'{pfx}dg-shouldfail[ ]*.*{sfx}')
+re_error = re.compile(f'{pfx}dg-error[ ]*.*{tgt}{sfx}')
+re_platform = re.compile('^[A-Za-z0-9*?_]+-[A-Za-z0-9*?_]+-[A-Za-z0-9*?_]+$')
+
+# Maps from known platforms to triples that LLVM will understand.
+# FIXME: The ia32 target probably does not always correspond to i386. Does it
+# means that it will be enabled on other non-X86 platforms?
+platforms = {'ia32': 'i386-*-*'}
+
+# Get the n-th level ancestor of the given file. The 1st level ancestor is
+# the directory containing the file. The 2nd level ancestor is the parent of
+# that directory and so on.
+def get_ancestor(f: str, n: int) -> str:
+    anc = f
+    for _ in range(0, n):
+        anc = os.path.dirname(anc)
+    return anc
+
+# Get the encoding of the file.
+def get_encoding(filepath: str) -> str | None:
+    with open(filepath, 'rb') as f:
+        return chardet.detect(f.read())['encoding']
+    return None
+
+# Get the lines in the file.
+def get_lines(filepath: str) -> list[str]:
+    lines = []
+    try:
+        encoding = get_encoding(filepath)
+        with open(filepath, 'r', encoding = encoding) as f:
+            lines = f.readlines()
+    except:
+        warning('Could not open file: {}', os.path.basename(filepath))
+    finally:
+        return lines
+
+# Collect the subdirectories of the gfortran directory which may contain tests.
+def get_subdirs(gfortran: str) -> list[str]:
+    regression = os.path.join(gfortran, 'regression')
+    torture = os.path.join(gfortran, 'torture')
+
+    subdirs = [regression]
+    for root, dirs, _ in os.walk(regression):
+        subdirs.extend([os.path.join(root, d) for d in dirs])
+    subdirs.append(torture)
+    for root, dirs, _ in os.walk(torture):
+        subdirs.extend([os.path.join(root, d) for d in dirs])
+    return subdirs
+
+# Strip any leading and trailing whitespace from the string as well as any
+# optional quotes around the string. Then split the string on whitespace and
+# return the resulting list.
+def qsplit(s: str) -> list[str]:
+    s = s.strip()
+    if s.startswith('"'):
+        s = s[1:]
+    if s.endswith('"'):
+        s = s[:-1]
+    return s.split()
+
+# Try to match the line with the regex. If the line matches, add the match
+# object to the MOUT list and return True. Otherwise, leave the MOUT list
+# unchanged and return False.
+def try_match(regex: re.Pattern, line: str, mout: list[re.Match[str]]) -> bool:
+    m = regex.search(line)
+    if m:
+        mout.append(m)
+        return True
+    return False
+
+# Count the number of elements in the list that match satisfy the predicate.
+def count_if(l, predicate):
+    return sum(1 for e in l if predicate(e))
+
+# Print a message. This is only around to save a bit of typing.
+def printf(fmt: str, *args) -> None:
+    print(fmt.format(*args))
+
+# Print a message in verbose mode.
+def message(fmt: str, *args) -> None:
+    printf(fmt, *args)
```

Comment:
Should this be controlled by some verbose flag?

Reply:
This is an obsolete comment. At some point, I had a verbose flag, but I removed it because it seemed unnecessary.